test/services/company/company.service.test.ts

```diff
@@ -87,4 +97,86 @@ describe('CompanyService', () => {
+
+      const error: Optional<string> = await service.validateCompanyDetails(company, TOKEN)
+
+      assert.isNull(error)
+    })
+
+    it('should return error when company is not of closeable type', async () => {
+
+      const companyOfficers: DirectorDetails[] = [generateDirectorDetails()]
+      const company: CompanyDetails = generateCompanyDetails()
+      company.companyNumber = COMPANY_NUMBER
+      company.companyName = 'Some company name'
+      company.companyStatus = 'active'
+      company.companyType = 'chicken'
+
+      when(officersService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve(companyOfficers)
+
+      const error: Optional<string> = await service.validateCompanyDetails(company, TOKEN)
+
+      assert.include(error, 'Company type of chicken cannot be closed via this service.')
+    })
+
+    it('should return error when company is not active', async () => {
+
+      const companyOfficers: DirectorDetails[] = [generateDirectorDetails()]
+      const company: CompanyDetails = generateCompanyDetails()
+      company.companyNumber = COMPANY_NUMBER
+      company.companyName = 'Some company name'
+      company.companyStatus = 'inactive'
+      company.companyType = ClosableCompanyType.LLP
+
+      when(officersService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve(companyOfficers)
+
+      const error: Optional<string> = await service.validateCompanyDetails(company, TOKEN)
+
+      assert.equal(error, 'The company is not currently active and cannot be closed.')
+    })
+
+    it('should return error when company is an overseas company', async () => {
+
+      const companyOfficers: DirectorDetails[] = [generateDirectorDetails()]
+      const company: CompanyDetails = generateCompanyDetails()
+      company.companyNumber = OverseasCompanyPrefix.FC + COMPANY_NUMBER
+      company.companyName = 'Some company name'
+      company.companyStatus = 'active'
+      company.companyType = ClosableCompanyType.LLP
+
+      when(officersService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve(companyOfficers)
+
+      const error: Optional<string> = await service.validateCompanyDetails(company, TOKEN)
+
+      assert.equal(error, 'This is an overseas company, and cannot be closed using this service.')
+    })
+
+    it('should return error when company has no active directors', async () => {
+
+      const companyOfficers: DirectorDetails[] = []
+      const company: CompanyDetails = generateCompanyDetails()
+      company.companyNumber = COMPANY_NUMBER
+      company.companyName = 'Some company name'
+      company.companyStatus = 'active'
+      company.companyType = ClosableCompanyType.LLP
+
+      when(officersService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve(companyOfficers)
+
+      const error: Optional<string> = await service.validateCompanyDetails(company, TOKEN)
+
+      assert.equal(error, 'The company has no active members / directors.')
```

Comment:
shouldn't this be dynamic based on company type? One to confirm with Jayti?

Reply:
Updated